connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -152,12 +166,23 @@ public void stop() {
         stopRequestedLatch.countDown();
         synchronized (this) {
             if (finishedStart)
-                task.stop();
+                tryStop();
             else
                 startedShutdownBeforeStartCompleted = true;
         }
     }
 
+    private void tryStop() {
```

Comment:
I know this is currently only called within a synchronized block, but this method implementation requires that be the case. What do you think about making this method synchronized, removing the synchronized block around the call in `close()` (lines 141-143 above), and leaving the other synchronized blocks as-is. I know that's overkill, but it's actually no more overhead and it's safer in case somebody in the future calls `tryStop()` outside of a synchronized block.